### PR TITLE
Use String Enum for preferences keys

### PR DESCRIPTION
**Pull request by windymilla**
Previous string keys were open to typos, since preferences are used across many different source files. Using `StrEnum` means it is no longer possible to set/query with a mistyped key.

In addition, a check has been added to  `initialize_preferences` to ensure that every preference has had a default set.

### CODE REVIEW

**Comment by windymilla** [comment]
I didn't do similar work the "bin" .json files, because that uses a `TypedDict`, which is only accessed in one file, in the `interpret_bin` and `create_bin` methods, unlike prefs which are used throughout the code. However,  maybe it would be better to use the StrEnum method, to remove the need for the duplication between lines 41-45 and 55-59 in `file.py`. Do you have a view on that, @cpeel?

**Review by cpeel — APPROVED** [review]
Very nice. We need to do something like this for the DP code too.

**Comment by cpeel** [comment]
> However, maybe it would be better to use the StrEnum method, to remove the need for the duplication between lines 41-45 and 55-59 in `file.py`. Do you have a view on that, @cpeel?

I don't think either approach is wrong. I would keep what you have since it's just used within one file and not over-engineer it.

**Comment by windymilla** [comment]
Testing notes for @srjfoo:
Should be no change in behavior, although the old GGprefs file will be invalid because the key names have change. 
If you wanted, you could try quitting, then manually editing the GGprefs file, then reloading, to see what happens if the user tries to fiddle with the settings and gets something wrong. I'm not too worried about it though because that's on them really.

**Review by srjfoo — APPROVED** [review]

File: src/guiguts/application.py [hunk 1]
```diff
@@ -252,25 +252,29 @@ def initialize_preferences(self) -> None:
...
+        preferences.set_default(PrefKey.SEARCHDIALOGREVERSE, False)
+        preferences.set_default(PrefKey.SEARCHDIALOGMATCHCASE, False)
+        preferences.set_default(PrefKey.SEARCHDIALOGWHOLEWORD, False)
+        preferences.set_default(PrefKey.SEARCHDIALOGWRAP, True)
+        preferences.set_default(PrefKey.SEARCHDIALOGREGEX, False)
+        preferences.set_default(PrefKey.DIALOGGEOMETRY, {})
+        preferences.set_default(PrefKey.ROOTGEOMETRY, "800x400")
+        preferences.set_default(PrefKey.DEFAULTLANGUAGES, "en")
+
+        # Check all preferences have a default
```
[the remark this answers is not in the document]
Comment (windymilla) [reply src/guiguts/application.py:275]: That's correct

File: src/guiguts/preferences.py [hunk 2]
```diff
@@ -13,6 +14,26 @@
 logger = logging.getLogger(__package__)
 
 
+class PrefKey(StrEnum):
```
[the remark this answers is not in the document]
Comment (windymilla) [reply src/guiguts/preferences.py:17]: No logical order. They are sort of grouped together by function, but not in any particular order. As the number increases, we may want to do that.

**Pull request merged**